pubnub-gson/pubnub-gson-impl/src/integrationTest/java/com/pubnub/api/integration/PublishIntegrationTests.java

```diff
@@ -107,6 +110,52 @@ public void testPublishMessageHistory() throws PubNubException {
         Awaitility.await().atMost(Durations.TEN_SECONDS).untilTrue(success);
     }
 
+    @Test
+    public void testPublishMessageWithOverrideUuid() throws PubNubException {
+        final AtomicBoolean success = new AtomicBoolean();
+        final String expectedChannel = randomChannel();
+        final JsonObject messagePayload = new JsonObject();
+        final String expectedUser = PubNub.generateUUID();
+
+        try {
+            messagePayload.addProperty("name", "joe");
+            messagePayload.addProperty("age", 48);
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+
+        final JsonObject whatToExpect = messagePayload;
+
+        PNConfiguration overrideConfig = PNConfigurationOverride.from(pubNub.getConfiguration()).setUserId(new UserId(expectedUser)).build();
```

Comment:
I can do like this:
        PNConfiguration overrideConfig = PNConfigurationOverride.from(pubNub.getConfiguration())
                .setUserId(new UserId(expectedUser))
                .presenceTimeout(0)
                .heartbeatInterval(4)
                .cacheBusting(true)
                .build();
Do we want to allow modification of all methods from PNConfiguration#Builder?